cookbooks/bcpc-hadoop/recipes/datanode.rb

```diff
@@ -182,6 +202,10 @@
   Chef::Application.fatal!("You have fewer #{node[:bcpc][:hadoop][:disks]} than #{node[:bcpc][:hadoop][:hdfs][:failed_volumes_tolerated]}! See comments of HDFS-4442.")
 end
 
+link "/etc/init.d/hadoop-yarn-nodemanager" do
```

Comment:
Annoyingly the relationship and order between the following is really difficult at this time. I'm thinking a definition or some other "package" would be good. The following steps are used to install a component at the moment:
1. Package install
2. `hdp-select`
3. Setting up symlinks
4. Starting a service (and particularly subscribes upon anything above)

The problem which I have often hit is as @http-418 reported:
````
No such file or directory - /etc/init.d/<foo service> <action>
````
As perhaps a `service` resource was subscribing on a package or `hdp-select` but thus triggered before the link was created. Or similarly, if the subscribe was `:delayed` and triggered due to a Chef run failure and the execution of all queued delayed executions. 

Due to the above, and again wanting to refactor this into components I am inclined to leave this in place for now.